lib/AST/Type.cpp

```diff
@@ -809,13 +809,16 @@ Type TypeBase::getWithoutParens() {
 Type TypeBase::replaceCovariantResultType(Type newResultType,
                                           unsigned uncurryLevel) {
   if (uncurryLevel == 0) {
-    if (auto objectType = getOptionalObjectType()) {
+    bool isLValue = is<LValueType>();
+
+    auto loadedTy = getRValueType();
```

Comment:
Just an aside: I think most usages of getRValueType() should be replaced by getWithoutSpecifierType(). getRValueType() also recursively walks into tuples and so on, and we don't usually (or ever?) need to do that.

Reply:
Thanks, I will change that!

Reply:
Done!